src/IO/S3Common.cpp

```diff
@@ -175,7 +172,7 @@ void AuthSettings::updateFrom(const AuthSettings & from)
         expiration_window_seconds = from.expiration_window_seconds;
 
     if (from.no_sign_request.has_value())
-        no_sign_request = *from.no_sign_request;
+        no_sign_request = from.no_sign_request;
```

Comment:
what's the difference here?

Reply:
It is the same actually. I just made it consistent with other code below.